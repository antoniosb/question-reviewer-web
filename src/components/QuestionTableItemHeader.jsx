import React from 'react';
import { Icon } from 'antd'
import formatDateTime from '../utils/date'

export default class QuestionTableItemHeader extends React.PureComponent {
  render() {
    const { item } = this.props
    return (
      <p className="question-table-item-header">
        Questão #{item.id} - <Icon type="user" /> {item.user.login}
        <div><Icon type="clock-circle-o" /> Última atualização em {formatDateTime(item.updated_at)}</div>
      </p>
    )
  }
}