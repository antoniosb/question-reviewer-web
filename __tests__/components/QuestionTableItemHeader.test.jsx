import React from 'react'
import QuestionTableItemHeader from '../../src/components/QuestionTableItemHeader'

describe('Components > QuestionTableItemHeader', () => {
  test('Render', () => {
    const component = shallow(<QuestionTableItemHeader item={{
      user: { login: 'teste' },
      id: 1,
      updated_at: new Date(2000, 11, 1, 12, 1, 1).toISOString()
    }} />)

    expect(component).toMatchSnapshot()
  })
})