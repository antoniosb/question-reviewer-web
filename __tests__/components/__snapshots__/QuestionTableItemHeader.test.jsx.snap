// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > QuestionTableItemHeader Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionTableItemHeader
    item={
      Object {
        "id": 1,
        "updated_at": "2000-02-01T10:00:00.000Z",
        "user": Object {
          "login": "teste",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Questão #",
        1,
        " - ",
        <Icon
          type="user"
        />,
        " ",
        "teste",
        <div>
          <Icon
            type="clock-circle-o"
          />
           Última atualização em 
          01/02/2000 às 05:00
        </div>,
      ],
      "className": "question-table-item-header",
    },
    "ref": null,
    "rendered": Array [
      "Questão #",
      1,
      " - ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "type": "user",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " ",
      "teste",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              type="clock-circle-o"
            />,
            " Última atualização em ",
            "01/02/2000 às 05:00",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "type": "clock-circle-o",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Última atualização em ",
          "01/02/2000 às 05:00",
        ],
        "type": "div",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Questão #",
          1,
          " - ",
          <Icon
            type="user"
          />,
          " ",
          "teste",
          <div>
            <Icon
              type="clock-circle-o"
            />
             Última atualização em 
            01/02/2000 às 05:00
          </div>,
        ],
        "className": "question-table-item-header",
      },
      "ref": null,
      "rendered": Array [
        "Questão #",
        1,
        " - ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": "user",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        "teste",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                type="clock-circle-o"
              />,
              " Última atualização em ",
              "01/02/2000 às 05:00",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "type": "clock-circle-o",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Última atualização em ",
            "01/02/2000 às 05:00",
          ],
          "type": "div",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
